### PR TITLE
icp: rip out everything we don't use

### PR DESCRIPTION
**Pull request by robn**
### Motivation and Context

After #16135, I started looking around the ICP module for gremlins, and goodness me there's a lot of _stuff_ in there. I'd like to simplify it all a great deal, but for now, it seems reasonable to remove everything we're not using, to reduce surface area and make it easier to understand what's there. This PR does just that!

### Description

See commits. I ran through removing everything that could just be cut with no or minimal code changes.

### How Has This Been Tested?

Compile checked on Linux and FreeBSD, full ZTS run completed.

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:
- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by adamdmoss** [comment]
I always wondered where ICP came from - looks like it's a lump of stuff from illumos.  OpenZFS is far beyond the point where it's practical to take upstream updates from illumos, right, which would be the only good reason to keep this code reasonably unmolested?  In which case :+1:

File: lib/libicp/Makefile.am [hunk 1]
```diff
@@ -31,7 +31,6 @@ nodist_libicp_la_SOURCES = \
 	module/icp/illumos-crypto.c \
 	module/icp/io/aes.c \
 	module/icp/io/sha2_mod.c \
-	module/icp/io/skein_mod.c \
```
Comment (allanjude) [inline lib/libicp/Makefile.am:37]: shouldn't this removal of skein_mod.c go in the other commit?

Comment (robn) [reply lib/libicp/Makefile.am:37]: Good spot, fixed.

**Comment by robn** [comment]
@adamdmoss yeah, that's my understanding of the history, and yeah, same conclusion here - we don't get anything much from it being this way, but we pay for maintenance overhead.

**Review by behlendorf — APPROVED** [review]
I'm all for slimming this down to what we actually need.  This looks good, although I'm curious how you ended up identifying the unsued functionality.  Was it all through code inspection?  It'd be interesting to see what a code coverage tool reports about how much of the remaining code the ZTS exercises.

**Comment by bghira** [comment]
icp was added during the crypto feature work i believe to avoid GPL libraries.

**Comment by robn** [comment]
Last push updates author and sponsor tags.

@behlendorf yes, just reading code and cutting things out, keeping the build running. I deliberately limited myself to removing things, keeping refactoring to a minimum.

**Comment by mcmilk** [comment]
@robn - thanks for the icp cleaning up patch series 👍

**Pull request closed without merging**